pkg/volume/glusterfs/glusterfs.go

```diff
@@ -331,7 +332,8 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 	// Refer to backup-volfile-servers @ http://docs.gluster.org/en/latest/Administrator%20Guide/Setting%20Up%20Clients/
 
 	if (len(addrlist) > 0) && (addrlist[0] != "") {
-		ip := addrlist[0]
+		// we'll pick a random address from the list to prevent a mount-storm
+		ip := addrlist[rand.Intn(len(addrlist))]
```

Comment:
IMO, this patch may not be needed, the `addrlist` itself is formed from `range` which is 'random' . Ideally this shouldnt be the case that, its always `same IP` get populated in all the attempts. Are you seeing a behaviour that always one IP is used ? if yes how many iterations were performed at your end ?

Reply:
@adampl there is little advantage in selecting the local node's IP. The initial request only fetches the volume layout (small text file, also called `.vol` file), after which the mount process will talk to all the Gluster bricks directly.

Gluster users the brick that responds fastest for reading data, but writing data is synchronous to all bricks that form a replica set.

Reply:
Thanks for the explanation.